### DISCUSSION TITLE
What is the proper way to test Hue?

### DISCUSSION THREAD

**Discussion by byungnam** [cloudera/hue#2475] [Questions] [answered]
I'm currently running `hue test unit` and a little bit customized `tools/jenkins/jenkins.sh` (which calls `hue test all --with-xunit --with-cover` inside) for tests.

Is this enough to test Hue? Or is it too much?

I'm asking this because I can successfully run `hue test unit` but in `tools/jenkins/jenkins.sh` (i.e. `hue test all --with-xunit --with-cover`), it shows a great amount of failures including a bunch of python2 python3 string to bytes incompatibility errors, assertion failures, temp file not removed errors, temp file not created errors, etc.

How does CI doing tests? Do they cover `hue test all`? If not, is it enough to do unit test only?

**Comment by romainr** [comment, marked as answer]
I suppose this is about the API tests: https://docs.gethue.com/developer/development/#running-the-api-tests

Integration tests are not very supported as of today. There is a plan to have the proper CI booting the service to test via a container for that in https://github.com/gethue/compose at some point.

In the meantime it is recommended to focus on clear unit tests with mocks (pretty efficient strategy so far):

https://github.com/cloudera/hue/blob/master/.circleci/config.yml#L162
